scripts/fgi/run

```diff
@@ -0,0 +1,204 @@
+import getpass, os, subprocess, sys
+
+from kube import (
+    kube_init_context,
+    kube_select_cluster,
+    get_cluster_context,
+    kube_wait_job,
+    create_forge_job,
+)
+
+TAG = ""
+WORKSPACE = ""
+DEFL_TIMEOUT_SECS = 2700  # Default timeout is 45 mins
+USER = getpass.getuser()  # Use the current user for naming
+OUTPUT_TEE = os.getenv("FGI_OUTPUT_LOG", tempfile.mkstemp()[1])
+
+HEADER = "\033[95m"
+OKBLUE = "\033[94m"
+OKGREEN = "\033[92m"
+WARNING = "\033[93m"
+FAIL = "\033[91m"
+RESTORE = "\033[0m"
+
+# build the arg parser and return a tuple of (fgi args, forge args)
+def build_argparser():
+    parser = argparse.ArgumentParser(
+        description="Entrypoint for the Forge unified testing framework"
+    )
+    parser.add_argument(
+        "--timeout-secs",
+        default=DEFL_TIMEOUT_SECS,
+        help="Timeout for the Forge pod in seconds",
+    )
+    parser.add_argument(
+        "--workspace",
+        "-W",
+        help="Workspace or kubernetes cluster to run Forge on, rather than picking one at random",
+    )
+    parser.add_argument(
+        "--env",
+        "-E",
+        action="append",
+        default=[],
+        help="Extra environment variables to pass to Forge",
+    )
+    build_group = parser.add_mutually_exclusive_group(required=True)
+    build_group.add_argument("--tag", "-T", help="Image tag to use in kubernetes Forge tests")
+    build_group.add_argument("--pr", "-p", help="PR to build images from for kubernetes Forge tests")
+    build_group.add_argument(
+        "--local-swarm",
+        "-L",
+        action="store_true",
+        help="Run Forge tests locally instead of on a kubernetes cluster",
+    )
+    return parser.parse_known_args()
+
+
+def get_grafana_url(cluster_name):
+    grafana_url_pattern = "http://mon.CLUSTERNAME.aws.hlw3truzy4ls.com"
+    return grafana_url_pattern.replace("CLUSTERNAME", cluster_name)
+
+
+def cli_tool_installed(tool_name):
+    ret = subprocess.run(
+        ["which", tool_name], stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL
+    )
+    return ret.returncode == 0
+
+
+# ================ Parse the args ================
+args, forge_args = build_argparser()
+
+# build and push the images to be used, since an image tag
+# was not specified explicitly
+TAG = args.tag
+if not args.tag:
+    if args.pr:  # codebuild using a PR
+        ret = subprocess.call(
+            ["aws", "codebuild", "list-projects"], stdout=subprocess.DEVNULL
+        )
+        if ret != 0:
+            print(f"{FAIL}Failed to access codebuild. Try aws-mfa?{RESTORE}")
+            sys.exit(1)
+        subprocess.call(
+            ["./docker/build-aws.sh", "--build-forge", "--version", f"pull/{args.pr}"]
+        )
+        TAG = f"dev_{USER}_pull_{args.pr}"
+        print(
+            f"**TIP Use ./scripts/fgi -T {TAG} <...> to restart this run with the same tag without rebuilding it"
+        )
+
+# ================ Test setup ================
```

Comment:
I have to say: smoking hot nice drawing! Now we have our branding and big logo same as some luxury L and G words. hahah